internal/validators/validate.go

```diff
@@ -0,0 +1,80 @@
+package validators
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"unicode"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/stellar/go/strkey"
+)
+
+func NewValidator() *validator.Validate {
+	validate := validator.New()
+	_ = validate.RegisterValidation("public_key", publicKeyValidation)
+	validate.RegisterAlias("not_empty", "required")
+	return validate
+}
+
+func publicKeyValidation(fl validator.FieldLevel) bool {
+	addr := fl.Field().String()
+	return strkey.IsValidEd25519PublicKey(addr) || strkey.IsValidMuxedAccountEd25519PublicKey(addr)
+}
+
+func ParseValidationError(errors validator.ValidationErrors) map[string]interface{} {
+	fieldErrors := make(map[string]interface{})
+	for _, err := range errors {
+		fieldErrors[getFieldName(err)] = msgForFieldError(err)
+	}
+	return fieldErrors
+}
+
+// msgForFieldError gets the message for the given validation error (tag).
+func msgForFieldError(fieldError validator.FieldError) string {
+	switch fieldError.Tag() {
+	case "required":
+		return "This field is required"
+	case "not_empty":
+		return "This field cannot be empty"
+	case "public_key":
+		return "Invalid public key provided"
+	case "oneof":
+		params := strings.Join(strings.Split(fieldError.Param(), " "), ", ")
+		return fmt.Sprintf("Unexpected value %q. Expected one of the following values: %s", fieldError.Value(), params)
+	case "gt":
+		if fieldError.Kind() == reflect.Slice || fieldError.Kind() == reflect.Array {
+			return "Should have at least 1 element"
+		}
+		return fmt.Sprintf("Should be greater than %s", fieldError.Param())
+	case "gte":
+		return fmt.Sprintf("Should be greater than or equal %s", fieldError.Param())
```

Comment:
I didn't know gt could be used with slices and arrays. In any case, the get and gt messages re not consistent in case of arrays.

Reply:
Also, is `Should have at least 1 element` what you expect to return when using gt with arrays? Isn't `fieldError.Param()` relevant for that case?

Reply:
Thank you for this! I'll change it so both will have the same behavior for slices. Also, I'm going to use the param here.